src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -1,15 +1,51 @@
 package ch.fhnw.algd1.converters.binary;
 
 public class BinConverter {
-	public static String toString(int x) {
-		// TODO expect x to be in range [-128, 127], return string with 8 binary
-		// digits representing x in 2-complement
-		return "00000000";
-	}
-
-	public static int fromString(String text) {
-		// TODO expect text to contain 8 binary digits, parse to int value in
-		// 2-complement
-		return 0;
-	}
+public static String toString(int x) {
+
+/*
+    var str = Integer.toBinaryString(x & 0xFF); // it's now unsigned
+
+    var zeros = new String(new char[8 - str.length()]).replace('\0', '0');
+
+    return zeros + str;
+    */
+    
+ 
+    StringBuilder value = new StringBuilder();
+    if (x < 0) x += 256;
+    
+    int i = 1 << 7;
+    while (i >= 1) {
+        if (x >= i) {
+            value.append("1");
+            x -= i;
+        } else {
+            value.append("0");
+        }
+        i /= 2;
+    }
+    return value.toString();
+}
+
+public static int fromString(String text) {
+    int value = 1;
+    int result = 0;
+    char[] chars = text.toCharArray();
+    for (int i = 0; i < chars.length; i++) {
+        if (chars[chars.length - 1 - i] == '0') {
+            value *= 2;
+            continue;
+        }
+        if (i + 1 == chars.length) {
+            value = -value;
+            result += value;
+            continue;
+        }
+        result += value;
+        value *= 2;
```

Comment:
Hier sehe ich nun überhaupt keinen Grund, warum man nicht die Programmstruktur auch durch die Struktur des Textes sichtbar machen soll:
```suggestion
        if (chars[chars.length - 1 - i] == '0') {
            value *= 2;
        } else if (i + 1 == chars.length) {
            value = -value;
            result += value;
        } else {
            result += value;
            value *= 2;
        }
```